CMakeLists.txt

```diff
@@ -167,6 +167,11 @@ else()
   set(MLX_BUILD_ACCELERATE OFF)
 endif()
 
+find_package(MPI)
```

Comment:
Is this still needed now that you do it dynamically?

Reply:
Well it depends :-) We can skip it and then I 'd add an mpi.h defining all the functions that I am using. It is basically finding the header.

Reply:
Functions are fine actually, the types need defining.

Reply:
I see, let's leave it then! It should still build without MPI installed which is good.